src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -267,6 +267,54 @@ namespace OpenLoco::World::MapGenerator
         }
     }
 
+    // 0x004611DF
+    static void sub_4611DF()
+    {
+        for (auto pos : World::getDrawableTileRange())
+        {
+            auto tile = TileManager::get(pos);
+            auto* surface = tile.surface();
+
+            if (surface == nullptr)
+                continue;
+
+            if (surface->isIndustrial())
+            {
+                auto* landObj = ObjectManager::get<LandObject>(surface->terrain());
+                if (landObj->hasFlags(LandObjectFlags::unk0))
+                {
+                    if (surface->water())
+                    {
+                        auto waterHeight = surface->waterHeight();
+                        if (surface->slope())
+                            waterHeight -= kSmallZStep;
+
+                        if (waterHeight > surface->baseZ())
+                        {
+                            if (surface->terrain() & 0xE)
+                            {
+                                surface->setVar6SLR5(0);
+                            }
+                        }
+                        else
+                        {
+                            surface->setVar6SLR5(landObj->var_03 - 1);
+                        }
+                    }
+                    else
+                    {
+                        surface->setVar6SLR5(landObj->var_03 - 1);
```

Comment:
any simple way to combine these two instances of `surface->setVar6SLR5(landObj->var_03 - 1);` could yank out the if logic into a standalone function.